src/java/org/apache/cassandra/service/StorageService.java

```diff
@@ -2296,105 +2296,117 @@ private void handleStateNormal(final InetAddress endpoint, final String status)
         // Order Matters, TM.updateHostID() should be called before TM.updateNormalToken(), (see CASSANDRA-4300).
         UUID hostId = Gossiper.instance.getHostId(endpoint);
         InetAddress existing = tokenMetadata.getEndpointForHostId(hostId);
-        if (replacing && isReplacingSameAddress() && Gossiper.instance.getEndpointStateForEndpoint(DatabaseDescriptor.getReplaceAddress()) != null
-            && (hostId.equals(Gossiper.instance.getHostId(DatabaseDescriptor.getReplaceAddress()))))
-            logger.warn("Not updating token metadata for {} because I am replacing it", endpoint);
-        else
-        {
-            if (existing != null && !existing.equals(endpoint))
+        boolean acquiredTokenLock = false;
```

Comment:
Should we synchronize this entire function?  Not sure if we ever expect more than one thread to ever enter here anyway, but might be safer to ensure that.

Reply:
There should only be a single thread for gossip, but I agree let's be defensive here.